From 4317cc7cf5709f6dc4b890a56a81495b1bd4de88 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Mon, 30 Aug 2021 10:29:33 +0300
Subject: [PATCH] [SW-36124] preboot: gaudi2: print test mode

System team requested to print OAM pins status at boot
One of them is test mode indication. Read GPIO pin 89

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I8b69a0751f19ba06835f60cf9f4a77d865d25237
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index 7ac7f4a..0c40e7b 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -375,6 +375,8 @@
 __weak u32 preboot_device_id_get(void) { return 0; }
 __weak void preboot_engines_init(void) {}
 __weak void preboot_read_straps(void) {}
+__weak void preboot_clear_wd_gpio(void) {}
+__weak void preboot_read_test_mode(void) {}
 
 void preboot_main(void)
 {
@@ -422,6 +424,7 @@
 	preboot_version();
 	fuse_version();
 	preboot_read_straps();
+	preboot_read_test_mode();
 	preboot_security_dynamic_vers_init(ver_off, sizeof(preboot_vers));
 
 	preboot_led_on();
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 782a8f7..2a6c5d2 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -10,6 +10,8 @@
 #include <psoc_global_conf_regs.h>
 #include <psoc_global_conf_masks.h>
 #include <psoc_gpio1_regs.h>
+#include <psoc_gpio3_regs.h>
+#include <psoc_gpio5_regs.h>
 #include <psoc_qspi_regs.h>
 #include <psoc_efuse_regs.h>
 #include <pcie_phy_regs.h>
@@ -653,3 +655,27 @@
 	for (i = 0; i < cfg_ctx->blocks; i++)
 		gaudi2_init_block_instances(i, cfg_ctx);
 }
+
+/*
+ * TEST_GPIO(89) is set through mmPSOC_GPIO5 BIT(9)
+ * Test Mode Indication
+ * 1: Normal operation
+ * 0: ASIC/GPU enter into electrical compliance mode
+ */
+#define TEST_MODE_GPIO		9
+void preboot_read_test_mode(void)
+{
+	u32 val, mask = BIT(TEST_MODE_GPIO);
+
+	/* configure as input */
+	regmap_writel(mask, mmPSOC_GPIO5_OUTENCLR);
+
+	val  = regmap_readl(mmPSOC_GPIO5_DATA);
+
+	if (val & mask)
+		preboot_puts_raw("Test mode indication: Normal operation");
+	else
+		preboot_puts_raw("Test mode indication: Compliance mode");
+
+	preboot_puts("\n");
+}
